homeassistant/components/matter/binary_sensor.py

```diff
@@ -34,60 +33,70 @@ async def async_setup_entry(
 class MatterBinarySensor(MatterEntity, BinarySensorEntity):
     """Representation of a Matter binary sensor."""
 
-    entity_description: MatterBinarySensorEntityDescription
-
-    @callback
-    def _update_from_device(self) -> None:
-        """Update from device."""
-        self._attr_is_on = self.get_matter_attribute_value(
-            # We always subscribe to a single value
-            self.entity_description.subscribe_attributes[0],
-        )
-
-
-class MatterOccupancySensor(MatterBinarySensor):
-    """Representation of a Matter occupancy sensor."""
-
-    _attr_device_class = BinarySensorDeviceClass.OCCUPANCY
-
     @callback
     def _update_from_device(self) -> None:
         """Update from device."""
-        value = self.get_matter_attribute_value(
-            # We always subscribe to a single value
-            self.entity_description.subscribe_attributes[0],
-        )
+        value: bool | uint | int | Nullable | None
+        value = self.get_matter_attribute_value(self._entity_info.primary_attribute)
+        if value in (None, NullValue):
+            value = None
+        elif value_convert := self._entity_info.measurement_to_ha:
+            value = value_convert(value)
+        self._attr_is_on = value
+
+
+# Discovery schema(s) to map Matter Attributes to HA entities
+DISCOVERY_SCHEMAS = [
+    # device specific: translate Hue motion to sensor to HA Motion sensor
```

Comment:
translate Hue motion sensor to HA Motion sensor

Reply:
remind for follow up cleanup PR